mythril/laser/ethereum/svm.py

```diff
@@ -114,6 +115,21 @@ def accounts(self) -> Dict[str, Account]:
         """
         return self.world_state.accounts
 
+    def set_standard_initial_state(
+        self, accounts: Dict[str, Account], ignore_addr=True
+    ):
+        initial_state = self.world_state.initial_state_account
```

Comment:
What's the reason for introducing a new satastructure? Both initial state, and initial_state_account in the world state class

Reply:
there is 1 var: initial_state_account, which will just have an initial account state and is propagated everywhere through reference everywhere.

Reply:
But depending on the trace through the constructor this account can be different

Reply:
Additionally this introduces some record of accounts using dicts while we already have a datamodel for account's

Reply:
How about we keep track of each transactions' initial and final world state. Then, when serializing you just take the state of the world before the first transaction that you are interested in. 
This is rather simple to add, whereas the current approach adds more complexity to the svm

wdyt

Reply:
it also seems that the transaction serialization is dependent on the structure of the account dicts, this dependency is not apparent and could easily break in the future if we would change or add variables

Reply:
`How about we keep track of each transactions' initial and final world state. Then, when serializing you just take the state of the world before the first transaction that you are interested in` But most of the data might redundant right?
`But depending on the trace through the constructor this account can be different`
It's a dict of all accounts(it seems to be a bad variable name). 
`this dependency is not apparent and could easily break in the future if we would change or add variables` If we change variables then yes, else no, I think that should be fine as the members are allowed access there, some tests might help here WDYT?

Reply:
It seems we already store those :man_facepalming:.